protocols/cifs/src/main/java/org/opennms/netmgt/poller/monitors/JCifsMonitor.java

```diff
@@ -148,18 +156,33 @@ public boolean accept(SmbFile smbFile, String s) throws SmbException {
         // Initializing TimeoutTracker with default values
         TimeoutTracker tracker = new TimeoutTracker(parameters, DEFAULT_RETRY, DEFAULT_TIMEOUT);
 
-        setJCifsTimeouts(tracker.getConnectionTimeout());
+        BaseContext baseCtx = null;
+        Properties jcifsProps = new Properties();
+        String toStr = "" + tracker.getConnectionTimeout();
+        jcifsProps.setProperty("jcifs.smb.client.soTimeout", toStr);
+        jcifsProps.setProperty("jcifs.smb.client.connTimeout", toStr);
+        jcifsProps.setProperty("jcifs.smb.client.responseTimeout", toStr);
+        jcifsProps.setProperty("jcifs.smb.client.sessionTimeout", toStr);
+        Configuration jcifsConfig = null;
+        try {
+            jcifsConfig = new PropertyConfiguration(jcifsProps);
+        } catch (CIFSException cifse) {
+            logger.error("Unable to configure CIFS timeout properties due to {}.", cifse.getMessage());
+            return PollStatus.unknown("Unable to configure CIFS timeout properties due to " + cifse.getMessage());
```

Comment:
I would continue ahead with the poll as opposed to returning `PollStatus.unknown`

Reply:
@j-white good point! Changed it to fall back to SingletonContext in that case.